code/modules/overmap/ship_consoles/_ship_console.dm

```diff
@@ -0,0 +1,75 @@
+/obj/machinery/computer/ship
+	name = "Broken Ship Console"
+	desc = "Please alert coders."
+	/// The instance of the simulated ship our console is connected to.
+	var/obj/structure/overmap/ship/simulated/current_ship
+	/// The sound to play when our console "turns on"
+	var/activate_sound = 'sound/machines/terminal_on.ogg'
+	/// The sound to play when our console "turns off"
+	var/deactivate_sound = 'sound/machines/terminal_off.ogg'
+	/// Current activation state
+	VAR_PRIVATE/activated = FALSE
+
+/obj/machinery/computer/ship/proc/get_active_uis()
+	PRIVATE_PROC(TRUE)
+	var/our_ref = "[REF(src)]"
+	return length(SStgui.open_uis_by_src[our_ref])
+
+/obj/machinery/computer/ship/proc/update_state()
+	SHOULD_NOT_SLEEP(TRUE)
+	var/toggle = get_active_uis()
+	var/prev = activated
+	if(!!activated != !!toggle) // If we are not in the state we should be in
+		activated = !activated
+	if(activated != prev)
+		playsound(src, activated ? activate_sound : deactivate_sound, 50)
```

Comment:
Is this is better than the old way of doing it? ~~Genuinely asking, I am not really in a position to check~~

Reply:
Oh, also, does it account for UIs opened by ghosts

Reply:
It doesn't really have the ability, without shitty snowflake code, to distingush between ghosts and mob/living unfortunately

Reply:
"Captain, why is our vessel accelerating towards the sun?"

Reply:
Ghosts can't interact with the console, only have it open. That won't be an issue